tests/test_s3api_multipart.sh

```diff
@@ -15,27 +15,112 @@
 # under the License.
 
 source ./tests/setup.sh
-source ./tests/test_aws_root_inner.sh
-source ./tests/util_file.sh
-source ./tests/util_multipart.sh
-source ./tests/util_tags.sh
+source ./tests/test_s3api_root_inner.sh
+source ./tests/util/util_file.sh
+source ./tests/util/util_multipart.sh
+source ./tests/util/util_tags.sh
 source ./tests/commands/get_object.sh
 source ./tests/commands/put_object.sh
 source ./tests/commands/list_multipart_uploads.sh
 
 # abort-multipart-upload
 @test "test_abort_multipart_upload" {
-  test_abort_multipart_upload_aws_root
+  local bucket_file="bucket-file"
+
+  run create_test_file "$bucket_file"
+  assert_success
+  # shellcheck disable=SC2154
+  run dd if=/dev/urandom of="$TEST_FILE_FOLDER/$bucket_file" bs=5M count=1
+  assert_success
+
+  run setup_bucket "s3api" "$BUCKET_ONE_NAME"
+  assert_success
+
+  run run_then_abort_multipart_upload "$BUCKET_ONE_NAME" "$bucket_file" "$TEST_FILE_FOLDER"/"$bucket_file" 4
+  assert_success
+
+  run object_exists "s3api" "$BUCKET_ONE_NAME" "$bucket_file"
+  assert_failure 1
 }
 
 # complete-multipart-upload
 @test "test_complete_multipart_upload" {
-  test_complete_multipart_upload_aws_root
+  local bucket_file="bucket-file"
+  run create_test_files "$bucket_file"
+  assert_success
+
+  run dd if=/dev/urandom of="$TEST_FILE_FOLDER/$bucket_file" bs=5M count=1
+  assert_success
+
+  run setup_bucket "s3api" "$BUCKET_ONE_NAME"
+  assert_success
+
+  run multipart_upload "$BUCKET_ONE_NAME" "$bucket_file" "$TEST_FILE_FOLDER"/"$bucket_file" 4
+  assert_success
+
+  run download_and_compare_file "s3api" "$TEST_FILE_FOLDER/$bucket_file" "$BUCKET_ONE_NAME" "$bucket_file" "$TEST_FILE_FOLDER/$bucket_file-copy"
+  assert_success
 }
 
 # create-multipart-upload
 @test "test_create_multipart_upload_properties" {
-  test_create_multipart_upload_properties_aws_root
+  local bucket_file="bucket-file"
+
+  local expected_content_type="application/zip"
+  local expected_meta_key="testKey"
+  local expected_meta_val="testValue"
+  local expected_hold_status="ON"
+  local expected_retention_mode="GOVERNANCE"
+  local expected_tag_key="TestTag"
+  local expected_tag_val="TestTagVal"
+
+  os_name="$(uname)"
+  if [[ "$os_name" == "Darwin" ]]; then
+    now=$(date -u +"%Y-%m-%dT%H:%M:%S")
+    later=$(date -j -v +15S -f "%Y-%m-%dT%H:%M:%S" "$now" +"%Y-%m-%dT%H:%M:%S")
+  else
+    now=$(date +"%Y-%m-%dT%H:%M:%S")
+    later=$(date -d "$now 15 seconds" +"%Y-%m-%dT%H:%M:%S")
```

Comment:
Does this need to be $now + 15 seconds?  Looks like it is missing the addition symbol.

Reply:
No, that's the linux format for adding time.  It works on my Linux machine.  The addition symbol actually subtracts 15 hours for some reason.